source/common/common/utility.cc

```diff
@@ -257,4 +260,12 @@ uint32_t Primes::findPrimeLargerThan(uint32_t x) {
   return x;
 }
 
+std::regex RegexUtil::parseRegex(const std::string& regex, std::regex::flag_type flags) {
+  try {
```

Comment:
Nit: I think ideally this would be caught by PGV (https://github.com/lyft/protoc-gen-validate) annotations. The language allows one to express regex validity (e.g. https://github.com/lyft/protoc-gen-validate/blob/6cd364ac742e088e22c2d04e90358ce48e1e33e5/validate/validate.proto#L478).

However, this is not yet implemented for C++, and what form the regexes checking should take is unclear (do we really want RE2?). 

@rodaine @akonradi for context.

Reply:
It's not clear to me that one can write a regular expression to validate that a string is a well-formed regular expression (given arbitrarily nested groups and depending on the library in use). But perhaps PGV will add a well-known rule for regular expressions in the future. I'll add a TODO.

Reply:
@zuercher I like that idea. Right now (since it's written in Go), it expects RE2 syntax, would need to do some work to support other formats.

Reply:
Umm, yes, I think I may have linked to the wrong thing. The idea is not to validate a regex with a regex, but to add support in PGV to validate that REs are valid.